README.md

```diff
@@ -341,6 +341,71 @@ userFetch.meta.response.headers.get('FOO')
 
 Do not attempt to read bodies directly from `meta.request` or `meta.response`. They are provided for metadata purposes only.
 
+## Setting defaults and hooking into internal processing
+
+It is possible to modify the various defaults used by React Refetch, as well as substitute in custom implementations of internal functions. A simple usecase would be to avoid repeating the same option for every fetch block:
+
+```jsx
+import { connect } from 'react-refetch'
+const refetch = connect.defaults({
+  credentials: 'include'
+})
+
+refetch(props => ({
+  userFetch: `/users/${props.userId}`
+}))(Profile)
+```
+
+A more advanced usecase would be to replace the `buildRequest` internal function to, for example, modify headers on the fly based on the URL of the request, or using advanced [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request) options:
+
+```jsx
+// api-connector.js
+import { connect } from 'react-refetch'
+import urlJoin from 'url-join'
+import { getPrivateToken } from './api-tokens'
+
+const baseUrl = 'https://api.example.com/'
+
+export default connect.defaults({
+  buildRequest: function (mapping) {
+    const options = {
+      method: mapping.method,
+      cache: 'force-cache',
+      referrer: 'https://example.com',
+      headers: mapping.headers,
+      credentials: mapping.credentials,
+      redirect: mapping.redirect,
+      body: mapping.body
+    }
+
+    if (mapping.url.match(/private/)) {
+      options.headers['X-Api-Token'] = getPrivateToken()
+    }
+
+    if (mapping.url.match(/listOfServers.json$/)) {
+      options.integrity = 'sha256-BpfBw7ivV8q2jLiT13fxDYAe2tJllusRSZ273h2nFSE='
+    }
+
+    return new Request(urlJoin(baseUrl, mapping.url), options)
```

Comment:
This example suggests there's no use case for providing a default `Request` which is not covered by changing `buildRequest`, seeing as you could just `return new MyCustomRequest(...)` at the end of your custom `buildRequest`. Is there a use case for replacing `Request` without replacing `buildRequest`?
We at least need a warning preventing the user from supplying both custom `Request` and custom `buildRequest`, since a custom `Request` would only be used by the default `buildRequest`